tests/shell/pipeline/commands/external.rs

```diff
@@ -117,6 +117,23 @@ mod stdin_evaluation {
 
         assert_eq!(stderr, "");
     }
+
+    /* Currently runs forever, but should not be the case
+    #[test]
+    fn does_not_block_indefinitely() {
+        let stdout = nu!(
+            cwd: ".",
+            pipeline(r#"
+                iecho yes
+                | chop
+                | chop
+                | first 1
+            "#
+        ));
+
+        assert_eq!(stdout, "ye");
+    }
+    */
```

Comment:
Added this test, but it still doesn't work. I'm hoping my series of PRs will lead to this working 🙂